go-controller/pkg/node/base_node_network_controller.go

```diff
@@ -0,0 +1,233 @@
+				klog.Info("System may be upgrading, falling back to legacy K8S Service via management port")
+				// add back legacy route for service via management port
+				link, err := util.LinkSetUp(types.K8sMgmtIntfName)
+				if err != nil {
+					return fmt.Errorf("unable to get link for %s, error: %v", types.K8sMgmtIntfName, err)
+				}
+				var gwIP net.IP
+				for _, subnet := range config.Kubernetes.ServiceCIDRs {
+					if utilnet.IsIPv4CIDR(subnet) {
+						gwIP = mgmtPortConfig.ipv4.gwIP
+					} else {
+						gwIP = mgmtPortConfig.ipv6.gwIP
+					}
+					err := util.LinkRoutesApply(link, gwIP, []*net.IPNet{subnet}, config.Default.RoutableMTU, nil)
+					if err != nil {
+						return fmt.Errorf("unable to add legacy route for services via mp0, error: %v", err)
+					}
+				}
+			}
+		}
+	}
+
+	// need to run upgrade controller
+	go func() {
+		ver, err := upgradeController.WaitForTopologyVersion(ctx, bnnc.stopChan, types.OvnCurrentTopologyVersion, 30*time.Minute)
+		if err != nil {
+			if !errors.Is(err, upgrade.ErrStopped) {
+				if config.OvnKubeNode.Mode == types.NodeModeFull && !bnnc.IsSecondary() {
+					klog.Fatalf("Failed to get topology version while waiting for it to be updated to %d, now %d: %v", types.OvnCurrentTopologyVersion, ver, err)
+				} else {
+					klog.Errorf("Failed to get topology version while waiting for it to be updated to %d, now %d: %v", types.OvnCurrentTopologyVersion, ver, err)
+				}
+			}
+			return
+		}
+		klog.Infof("Cluster topology version is now %d network %s", ver, bnnc.GetNetworkName())
+
+		// ensure CNI support for port binding built into OVN, as masters have been upgraded
+		if initialTopoVersion < types.OvnPortBindingTopoVersion && !isOvnUpEnabled && !config.OvnKubeNode.DisableOVNIfaceIdVer {
+			isOvnUpEnabled, err := util.GetOVNIfUpCheckMode()
+			if err != nil {
+				klog.Errorf("%v", err)
+			} else if isOvnUpEnabled {
+				klog.Infof("Detected support for port binding with external IDs")
+				if cniServer != nil {
+					cniServer.EnableOVNPortUpSupport()
+				}
+				atomic.StoreInt32(&bnnc.atomicOvnUpEnabled, 1)
+			}
+		}
+
+		if config.OvnKubeNode.Mode == types.NodeModeFull && !bnnc.IsSecondary() {
+			// upgrade complete now see what needs upgrading
+			// migrate service route from ovn-k8s-mp0 to shared gw bridge
+			if (initialTopoVersion < types.OvnHostToSvcOFTopoVersion && config.GatewayModeShared == config.Gateway.Mode) ||
+				(initialTopoVersion < types.OvnRoutingViaHostTopoVersion) {
+				if err := upgradeServiceRoute(bridgeName); err != nil {
+					klog.Fatalf("Failed to upgrade service route for node, error: %v", err)
+				}
+			}
+		}
+	}()
+	return nil
+}
+
+// checkForStaleOVSRepresentorInterfaces checks for stale OVS ports backed by Repreresentor interfaces,
+// derive iface-id from pod name and namespace then remove any interfaces assoicated with a sandbox that are
+// not scheduled to the node.
+func (bnnc *BaseNodeNetworkController) checkForStaleOVSRepresentorInterfaces() {
```

Comment:
@jcaamano and I were discussin gif we should use external-ids:iface-id-ver to detect stale OVS representor, then call it globally. Otherwise, the current way use iface-id which needs to know be network aware. Let's discuss it in next meeting. 

if so, we should do it in the default network controller? or NNCM? if it is latter, it would be a new NetworkControllerManager() healthcheck interface.

If we do it globally, we don't need Cleanup function either.

Reply:
If we can do it with just iface-id-ver and not have to worry about network details lets go that route. OVN has supported iface-id-ver for a long time now and we dont need to make it backwards compatible. We should probably just get rid of the option to disable it via config.